File: Leetcode/PY/Leetcode523-连续的子数组和.py
from typing import DefaultDict, List
from collections import defaultdict
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        if n < 2:
            return False
        re = 0
        mp = defaultdict(int)
        mp[0] = -1
        for i in range(n):
            re = (re + nums[i]) % k
            if re in mp:
                if i - mp[re] >= 2:
                    return True
            else:
                mp[re] = i
        return False

File: Leetcode/Rust/Leetcode523-连续的子数组和.rs
use std::collections::HashMap;
impl Solution {
    pub fn check_subarray_sum(nums: Vec<i32>, k: i32) -> bool {
        let n = nums.len();
        if n < 2 {
            return false;
        }
        let mut re = 0;
        let mut mp = HashMap::new();
        mp.insert(0, -1 as i32);
        for i in 0..n{
            re = (re + nums[i]) % k;
            if mp.contains_key(&re){
                let idx = mp[&re];
                if i as i32- idx >= 2{
                    return true;
                }
            }else{
                mp.entry(re).or_insert(i as i32);
            }
        }
        return false;
    }
}